### PR TITLE
Add `droid` server preset for the pretrained DreamZero DROID checkpoint

### PR DESCRIPTION
**Pull request by vertix**
## Summary
- Add a `droid` preset to the DreamZero inference server that serves the public pretrained
  `GEAR-Dreams/DreamZero-DROID` checkpoint (wan2.1), mirroring how openpi exposes a named `droid` config.
- Add a `droid` codec (320×180) — the pretrained DROID model asserts exactly that resolution, vs the
  176-tall codec the wan2.2 fine-tunes resize at load.
- The server now downloads HuggingFace repos at serve time via `snapshot_download` (the upstream VLA
  loader reads `model_path` as a local dir, so the repo must be fetched by our wrapper); `s3://`/local
  paths still go through `pos3`.
- Support latest-checkpoint discovery: a bare `s3://` run directory resolves to its newest `checkpoint-N`
  via the shared `positronic.utils.checkpoints.get_latest_checkpoint`, so `serve.sh` can take a run-dir
  instead of a pinned checkpoint path. A path already ending in `checkpoint-N`, a HuggingFace repo, or a
  local path passes through unchanged.
- The base `server` config no longer defaults `model_path` to the DROID repo — `droid` owns that identity.

Note: `uv.lock` is regenerated to add `huggingface-hub`; the repo pins no `uv` version, so a full
`uv lock` also re-normalizes some platform markers (equivalent, passes `uv sync --locked`).

## Test plan
- Probed both presets end-to-end against live H100 endpoints with `positronic/probe.py`: `serve` (wan2.2
  fine-tune) and `droid` (wan2.1 pretrained) each returned a 24-action chunk.
- Verified checkpoint resolution against real S3: a run dir resolves to the latest `checkpoint-N`; a pinned
  checkpoint, a HuggingFace repo, and a local path each pass through unchanged.
- `ruff check`/`ruff format` clean; `dreamzero/tests/test_codecs.py` + `utils/tests/test_checkpoints.py` pass.

### CODE REVIEW

File: positronic/vendors/dreamzero/server.py [hunk 1]
```diff
@@ -13,23 +13,39 @@
...
-DEFAULT_HF_REPO = 'GEAR-Dreams/DreamZero-DROID'
-
 
 def _dreamzero_root():
     return Path(__file__).parents[4] / 'dreamzero'
 
 
+def _download_checkpoint(model_path: str) -> Path:
+    """Local checkpoint dir for ``model_path``: an ``s3://`` URL or local path via pos3, else a HuggingFace repo."""
+    if model_path.startswith('s3://') or os.path.exists(model_path):
```
[the remark this answers is not in the document]
Comment (vertix) [reply positronic/vendors/dreamzero/server.py]: Fixed in fedffab: `_download_checkpoint` now `os.path.expanduser`s `model_path` before the `os.path.exists` check and passes the expanded path to `pos3.download`, so `--model_path=~/ckpt` routes through the local-path flow instead of the HuggingFace fallback.

File: positronic/vendors/dreamzero/codecs.py [hunk 2]
```diff
@@ -223,6 +223,10 @@ def dreamzero_action(tgt_joints_key: str, tgt_grip_key: str, num_joints: int):
 _action = dreamzero_action.override(tgt_joints_key='robot_command.joints', tgt_grip_key='target_grip')
 joints = codecs.compose.override(obs=dreamzero_obs, action=_action, fps=15.0)
 
+# The pretrained DROID model (wan2.1) asserts exactly 320x180 frames; the wan2.2 fine-tunes resize the
+# 176-tall codec output at load, so only the DROID-serving codec needs the taller image.
+droid = codecs.compose.override(obs=dreamzero_obs.override(image_size=(IMAGE_WIDTH, 180)), action=_action, fps=15.0)
```
Comment (vertix) [inline positronic/vendors/dreamzero/codecs.py:228]: should it be named after wan2.2, rather than droid? Or is it specific for droid checkpoint?
also, probably we should call it just joints_2_2, and the prev call as joints_2_1, I don't know. The names should carry the functionality, not the usage
